docs/contributing/setting_up_your_dev_environment.rst

```diff
@@ -0,0 +1,122 @@
+==========================================
+
+#FIXME: Walk through these instructions at least once to verify success.
+
+Prerequisites
+-------------------
+
+In order to contribute to Great Expectations, you will need the following:
+
+* A GitHub account - this is sufficient if you only want to contribute to the documentation.
+* If you want to contribute code, you will also need a working version of Git on your computer. Please refer to the [Git setup instructions](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) for your environment.
+* We also recommend going through the [SSH key setup process on GitHub](https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) for easier authentication.
+
+
+Fork and clone the repository
+------------------------------
+
+**1. Fork the Great Expectations repo**
+
+    * Go to the `Great Expectations repo on GitHub <https://github.com/great-expectations/great_expectations>`__.
+    * Click the ``Fork`` button in the top right. This will make a copy of the repo in your own GitHub account.
+    * GitHub will take you to your forked version of the repository.
+
+
+**2.  Clone your fork**
+
+    * Click the green ``Clone`` button and choose the SSH or HTTPS URL depending on your setup. 
+    * Copy the URL and run ``git clone <url>`` in your local terminal.
+    * Note: This will clone the ``develop`` branch of the great_expectations repo by default, not ``master``. (#FIXME: Which branch should I clone??)
+
+
+**3. Add the upstream remote**
+
+    * On your local machine, cd into the great_expectations repo you cloned in the previous step.
+    * Run: ``git remote add upstream git@github.com:great-expectations/great_expectations.git``
+    * This sets up a remote called ``upstream`` to track changes to the main branch.
+
+**4. Create a feature branch to start working on your changes.**
+
+    * Ex: ``git checkout -b feature/my-feature-name``
+    * Your branch should be named {#FIXME: agree on a convention}. Please see Style guide for details.
+
+
+Install python dependencies
+------------------------------
+
+**5. Create a new virtual environment**
+
+    * Make a new virtual environment (e.g. using virtualenv or conda), name it "great_expectations_dev" or similar.
+    * Ex: ``virtualenv great_expectations_dev``
+    * This is not required, but highly recommended.
+
+**6. Install dependencies from requirements-dev.txt**
+
+    *  to make sure you have the right libraries, then install great_expectations from the version you just forked:
+    * `pip install -r requirements-dev.txt`
+
+**7. Install great_expectations from your cloned repo**
+
+    * ``pip install -e .``
+    * ``-e`` will install in "editable" mode. This is not required, but is often very convenient as a developer.
+
+**8. Make sure you have GCC and Java installed and working on your machine.**
+
+    * #FIXME: I'm not sure how to do this or why it's important.
```

Comment:
Great question... something prompted the Java installation and I had problems with GCC, possibly while trying to run tests... but feel free to drop that line for now.